src/fheroes2/agg/agg_image.cpp

```diff
@@ -1473,47 +1486,94 @@ namespace fheroes2
 
                 break;
             }
-            case ICN::BUTTON_DIFFICULTY_ARCHIBALD: {
-                _icnVsSprite[id].resize( 2 );
+            case ICN::GOOD_CAMPAIGN_BUTTONS:
+            case ICN::EVIL_CAMPAIGN_BUTTONS: {
+                _icnVsSprite[id].resize( 10 );
 
-                const int32_t textWidth = 140;
-                fheroes2::Point releasedOffset;
-                fheroes2::Point pressedOffset;
-                getCustomNormalButton( _icnVsSprite[id][0], _icnVsSprite[id][1], true, textWidth, releasedOffset, pressedOffset );
+                if ( useOriginalResources() ) {
+                    const bool isEvilInterface = id == ICN::EVIL_CAMPAIGN_BUTTONS;
+                    const int originalIcnId = isEvilInterface ? ICN::CAMPXTRE : ICN::CAMPXTRG;
 
-                renderTextOnButton( _icnVsSprite[id][0], _icnVsSprite[id][1], gettext_noop( "DIFFICULTY" ), releasedOffset, pressedOffset, { textWidth, 16 },
-                                    fheroes2::FontColor::GRAY );
+                    // The evil buttons' released state are 2 pixels wider.
+                    const int offsetEvilX = isEvilInterface ? 2 : 0;
+                    // remove embedded shadows so that we can generate shadows with our own code later
+                    for ( uint32_t i = 0; i < 4; ++i ) {
```

Comment:
As in this loop we use `i` four times multiplying it by 2 we can do this multiply once:
```suggestion
                    for ( uint32_t i = 0; i < 8; i += 2 ) {
```
and replace all `2 * i` by `i` on lines 1502, 1504, 1511, 1513.

Doing this we will reduce computations and also make Cland-Tydy happy. :)